src/Simplify_Call.cpp

```diff
@@ -125,28 +127,39 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
             return a;
         }
 
-        const bool shift_left = op->is_intrinsic(Call::shift_left);
+        //const bool const_shift_left = op->is_intrinsic(Call::shift_left);
+        //const bool const_shift_right = true;
+        const bool const_shift_left = ((op->is_intrinsic(Call::shift_left) && can_prove(b >= 0)) ||
```

Comment:
@abadams might be best to comment on this, but this might be overkill at this point in the simplifier.

It might be best to just try to handle constant shifts (replacing them with Mul/Div), like the old code did (but only for unsigned shift amounts). 

If nothing else, it might be best to leave more aggressive simplification to a follow-up PR.

Reply:
Andrew already commented, going to resolve this and let Andrew's comment be the one for discussion.